inspirehep/utils/date.py

```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2016 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""Helpers for handling dates."""
+
+import six
+
+from dateutil.parser import parse
+
+from invenio_utils.date import datetime, strftime, strptime
+
+DATE_FORMATS_YEAR = ["%Y", "%y"]
+DATE_FORMATS_MONTH = [
+    "%Y-%m", "%Y %b", "%b %Y", "%Y %B", "%B %Y",
+    "%y-%m", "%y %b", "%b %y", "%y %B", "%B %y",
+]
+DATE_FORMATS_FULL = [
+    "%Y-%m-%d", "%d %m %Y", "%x", "%d %b %Y",
+    "%d %B %Y", "%d %b %y", "%d %B %y",
+]
+
+
+def create_valid_dates(dates):
+    """Add '01' as a value of missing month and/or day values."""
```

Comment:
Basically I'd propose that ES and formats are made compatible with partial dates such as `2016` and `2016-05`, so that we don't introduce fake information, that might impact further computations and presentations.